src/irc.cc

```diff
@@ -69,6 +70,22 @@ void addmessage(struct ctx *ctxptr, char *nick, const char *text) {
 		}
 		lastmsg->next = newmsg;
 	}
+	if(ctxptr->max_msg_lines>0) {
+		int msgcnt = 0;
+		newmsg = ctxptr->messages;
+		while(newmsg) {
```

Comment:
I'm wondering what happens if there's a case where there's no null pointer in `newmsg`. Rather than having to read through all the code, I prefer to see some kind of bounds check. This is how security vulnerabilities are created (although I'm not saying you created one, I'm just being conservative).